lib/task-graph.js

```diff
@@ -208,10 +208,15 @@ function taskGraphFactory(
                 waitingOn: taskData.waitOn,
                 ignoreFailure: taskData.ignoreFailure
             };
-
-            if (taskData.taskName) {
+            if (taskData.taskName && typeof(taskData.taskName) !== "undefined") {
```

Comment:
the condition never goto 'typeof(taskData.taskName) !== "undefined"' which after '&&' , right?